sqlitecluster/SQLite.cpp

```diff
@@ -256,79 +310,146 @@ bool SQLite::write(const string& query) {
     return true;
 }
 
-// --------------------------------------------------------------------------
 bool SQLite::prepare() {
     SASSERT(_insideTransaction);
+
+    // We lock this here, so that we can guarantee the order in which commits show up in the database.
+    commitLock.lock();
+
+    // Now that we've locked anybody else from committing, look up the state of the database.
+    string committedQuery, committedHash;
+    uint64_t commitCount = _commitCount.load();
+
     // Queue up the journal entry
-    _uncommittedHash = SToHex(SHashSHA1(_committedHash + _uncommittedQuery));
+    string lastCommittedHash = getCommittedHash();
+    _uncommittedHash = SToHex(SHashSHA1(lastCommittedHash + _uncommittedQuery));
+    _uncommittedTransaction = true;
     uint64_t before = STimeNow();
-    bool result = SQuery(_db, "updating journal", "INSERT INTO journal VALUES ( NULL, " + SQ(_uncommittedQuery) + ", " +
-                                                      SQ(_uncommittedHash) + " )");
+
+    // This is a hack to write to multiple journals from a single thread. It exists only for testing purposes, and
+    // should be removed when adding real multi-threaded writes.
+    int table = (SRandom::rand64() % (_minJournalTables + 2)) - 1;
+    _journalName = getJournalTableName(table);
+
```

Comment:
remove this blank line, or add a comment beneath it.  Only split code up if it's meaningful; if it's not meaningful enough to warrant a comment under it, don't split.